ConsoleApp/Program.cs

```diff
@@ -0,0 +1,228 @@
+        ReadKey();
+    }
+
+    static string Serialize((string firstName, string lastName, string phone) row) => $"{row.firstName},{row.lastName},{row.phone}";
+
+    static void AddNewContact()
+    {
+        Clear();
+
+        WriteLine("Enter first name:");
+        var firstName = ReadLine().Trim();
+
+        WriteLine("Enter last name:");
+        var lastName = ReadLine().Trim();
+
+        WriteLine("Enter phone:");
+        var phone = ReadLine().Trim();
+
+        File.AppendAllLines(filename, new[] { Serialize((firstName, lastName, phone)) });
+
+        WriteLine("Contact saved, press any key to continue");
+        ReadKey();
+    }
+
+    static void RemoveContact()
+    {
+        Clear();
+
+        WriteLine("Enter phone to remove:");
+        var phoneToRemove = ReadLine();
+
+        var contacts = ReadContacts(filename);
+        var newContacts = new (string, string, string)[contacts.Length];
+        Array.Copy(contacts, newContacts, contacts.Length);
+
+        var index = 0;
+
+        while (index < newContacts.Length)
+        {
+            if (newContacts[index].Item3 == phoneToRemove)
+            {
+                newContacts[index] = newContacts[^1];
+                Array.Resize(ref newContacts, newContacts.Length - 1);
+            }
+            else
+            {
+                ++index;
+            }
+        }
+
+        RewriteContacts(newContacts);
+
+        WriteLine($"{contacts.Length - newContacts.Length} Contact(s) removed, press any key to continue");
+        ReadKey();
+    }
+    static void SearchContact()
+    {
+        Clear();
+        var contacts = ReadContacts(filename);
+        WriteLine("Enter Contact First or Last Name:");
+        string searchKey = ReadLine();
+
+        int counter = 0;
+        foreach (var (firstName, lastName, phoneNumber) in contacts)
+        {
+            if (firstName.ToLower() == searchKey.ToLower() || lastName.ToLower() == searchKey.ToLower())
+            {
+                ShowRow((firstName, lastName, phoneNumber));
+                ++counter;
+            }
+        }
+        if (counter == 0)
+        {
+            WriteLine("No Contacts Found!");
+        }
+        else
+        {
+            WriteLine($"{counter} {(counter == 1 ? "contact was" : "contacts were")} found");
+        }
+        WriteLine("Press any key to continue...");
+        ReadKey();
+
+    }
+    static void UpdateContact()
+    {
+        Clear();
+        var contacts = ReadContacts(filename);
+        WriteLine("Provide Information About The Contact You Would Like To Update");
+        WriteLine("First Name:");
+        string firstName = ReadLine();
+        WriteLine("Last Name:");
+        string lastName = ReadLine();
+        bool isContactFound = false;
+
+        for (int i = 0; i < contacts.Length; i++)
+        {
+            if (contacts[i].Item1.ToLower() == firstName.ToLower() && contacts[i].Item2.ToLower() == lastName.ToLower())
```

Comment:
it's also an option but it's better to use
```suggestion
            if (string.Equals(contacts[i].Item1, firstName, StringComparison.OrdinalIgnoreCase) && ...)
```